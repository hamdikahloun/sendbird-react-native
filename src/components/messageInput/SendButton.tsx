import {Assets} from '@/assets';
import React from 'react';
import {Image, StyleSheet, TouchableOpacity} from 'react-native';

type SendButtonProps = {
  onSendButtonPressed?: () => void;
};

/**
 * SendButton - UI Component for send message
 *
 * See:
 *
 * [Send your first message](https://sendbird.com/docs/chat/v3/javascript/quickstart/send-first-message)
 * @example
 *
 * <SendButton onSendButtonPressed={onSendButtonPressed} />
 */
export const SendButton = ({onSendButtonPressed}: SendButtonProps) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onSendButtonPressed}>
      <Image source={Assets.send} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
    height: 48,
    width: 48,
  },
});
