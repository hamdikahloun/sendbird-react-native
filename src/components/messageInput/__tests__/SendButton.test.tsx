import React from 'react';
import {render} from '@/../testing';
import {SendButton} from '../SendButton';

describe('SendButton', () => {
  it('should render correctly', async () => {
    const screen = render(<SendButton />).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
