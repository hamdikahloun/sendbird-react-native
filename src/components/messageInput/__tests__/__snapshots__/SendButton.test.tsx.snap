// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendButton should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
      "width": 48,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/send/send.png",
      }
    }
  />
</View>
`;
